rc/extra/ghcid.kak

```diff
@@ -0,0 +1,129 @@
+# ghcid integration (loosely based on make.kak)
+#
+# Starts ghcid in the background. all commands are executed on the
+# client specified by the 'ghcidtoolsclient' option.
+#
+# Recommended way to use this is to define a command in your kakrc
+# to split off a new client for showing the ghcid output:
+#
+#   set-option global ghcidtoolsclient ghcid
+#   define-command ghcid %{
+#     iterm-new-vertical "rename-client ghcid; ghcid-start"
+#   %}
+#
+# And you may want to bind the next and previous error commands to nice keys,
+# e.g. here ",p", and ",n":
+#   map -docstring "jump to previous error" global user p ':ghcid-previous-error<ret>'
+#   map -docstring "jump to next error" global user n ':ghcid-next-error<ret>'
+
+# options
+
+declare-option -docstring "name of the client in which to jump" \
+  str ghcidclient
+
+declare-option -docstring "name of the client in which to process output" \
+  str ghcidtoolsclient
+
+declare-option -docstring "ghcid error regexp" \
+  str ghcid_error_pattern "[^:\n]+:\d+:\d+: error:$"
+
+declare-option -hidden int ghcid_current_error_line
+
+# commands
+
+define-command -params .. \
+  -docstring %{ghcid-start [<arguments>]: ghcid utility wrapper
+All optional arguments forwarded to ghcid} \
+  ghcid-start %{ %sh{
+    workdir=$(mktemp -d "${TMPDIR:-/tmp}"/kak-ghcid.XXXXXXXXX)
+    output=${workdir}/output
+    errors=${workdir}/errors
+    fifo=${workdir}/fifo
+    exec 2> ${errors}
+    touch ${output}
+    mkfifo ${fifo}
+    cmd="ghcid -S -n -o ${output} $@"
+    echo "starting: ${cmd}" > ${output}
```

Comment:
Using `echo` with a string that has arbitrary expansions is not portable, prefer using `printf`.